File: LeetCode/53. Maximum Subarray.py
#Author : @AdarshGaur

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        lmax , gmax = nums[0], nums[0]
        for i in range(1, len(nums)):
            lmax += nums[i]
            if lmax <= nums[i]:
                lmax = nums[i]
            gmax = max(lmax, gmax)
        return gmax

File: LeetCode/53. Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0], curSum = 0;
        for(int i=0; i<nums.length; i++){
            curSum += nums[i];
            maxSum = curSum > maxSum ? curSum : maxSum;
            curSum = curSum < 0 ? 0 : curSum;
        }
        return maxSum;
    }
}
